contracts/vaults/Claim.sol

```diff
@@ -33,43 +33,58 @@ contract Claim is Base {
         string calldata _descriptionHash)
     external
     onlyCommittee(_pid)
-    noSubmittedClaims(_pid) {
+    noSubmittedClaims(_pid)
+    {
         require(_beneficiary != address(0), "HVE04");
         // require we are in safetyPeriod
         // solhint-disable-next-line not-rely-on-time
         require(block.timestamp % (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) >=
         generalParameters.withdrawPeriod, "HVE05");
         require(_bountyPercentage <= bountyInfos[_pid].maxBounty, "HVE06");
-
-        submittedClaims[_pid] = SubmittedClaim({
+        uint256 claimId;
+        uint256 nonce;
+        do
+        {
+            claimId = uint256(keccak256(abi.encodePacked(_pid, block.number, nonce)));
+            nonce++;
+        } while (claimId == 0 || submittedClaims[claimId].beneficiary != address(0));
+        submittedClaims[claimId] = SubmittedClaim({
+            pid: _pid,
             beneficiary: _beneficiary,
             bountyPercentage: _bountyPercentage,
             committee: msg.sender,
             // solhint-disable-next-line not-rely-on-time
             createdAt: block.timestamp
         });
-        emit SubmitClaim(_pid, msg.sender, _beneficiary, _bountyPercentage, _descriptionHash);
+        activeClaims[_pid] = claimId;
+        emit SubmitClaim(
+            _pid,
+            claimId,
+            msg.sender,
+            _beneficiary,
+            _bountyPercentage,
+            _descriptionHash
+        );
     }
 
     /**
     * @notice Approve a claim for a bounty submitted by a committee, and transfer bounty to hacker and committee.
     * Called only by hats governance.
-    * @param _pid The pool id
+    * @param _claimId The claim ID
     */
-    function approveClaim(uint256 _pid) external onlyOwner nonReentrant {
-        require(submittedClaims[_pid].beneficiary != address(0), "HVE10");
-        BountyInfo storage bountyInfo = bountyInfos[_pid];
-        SubmittedClaim memory submittedClaim = submittedClaims[_pid];
-        delete submittedClaims[_pid];
-
-        IERC20Upgradeable lpToken = poolInfos[_pid].lpToken;
-        ClaimBounty memory claimBounty = calcClaimBounty(_pid, submittedClaim.bountyPercentage);
-        poolInfos[_pid].balance -= claimBounty.hacker
-                            + claimBounty.hackerVested
-                            + claimBounty.committee
-                            + claimBounty.swapAndBurn
-                            + claimBounty.hackerHatVested
-                            + claimBounty.governanceHat;
+    function approveClaim(uint256 _claimId) external onlyOwner nonReentrant {
+        SubmittedClaim memory submittedClaim = submittedClaims[_claimId];
+        require(submittedClaim.beneficiary != address(0), "HVE10");
```

Comment:
mb it is more straightforward to require here that `activeClaims[_claimId]`

(should come down to the same thing I guess, but seems more readable)

Reply:
i was worried it could be unsafe because that way approveClaim(0) would pass

Reply:
ah sorry it is me who was confused :)